### PR TITLE
More perf improvements

### PR DESCRIPTION
**Pull request by elizabeth-legros**
Adding in backing structure to CLDS for local searches without IDs (like searching for ports)

### CODE REVIEW

File: com.unity.sg2/Tests/Generation/GenerationTests.cs [hunk 1]
```diff
@@ -43,9 +43,9 @@ public static void Setup()
             graph.RebuildContextData(propertyKey.Name, GetTarget(), "UniversalPipeline", "SurfaceDescription", true);
             //graph.RebuildContextData(contextKey.Name, GetTarget(),  "UniversalPipeline", "SurfaceDescription", false);
 
-            CPGraphDataProvider.GatherProviderCPIO(GetTarget(), out var descriptors);
-            foreach (var descriptor in descriptors)
-                LogDescriptor(descriptor);
+            //CPGraphDataProvider.GatherProviderCPIO(GetTarget(), out var descriptors);
```
Comment (MattSutherlin) [inline com.unity.sg2/Tests/Generation/GenerationTests.cs:46]: Do we think this code might come back or should it be deleted?

**Review by MattSutherlin — APPROVED** [review]

**Review by brett-groshong — APPROVED** [review]

**Comment by bencloward** [comment]
These changes are working really well and make loading files and working in the tool quite a bit faster.  Because of this, I'm now working exclusively in this branch until the changes are merged.

**Review by bencloward — APPROVED** [review]

File: com.unity.sg2/Tests/Generation/GenerationTests.cs (hunk 1 above)
Comment (elizabeth-legros) [reply com.unity.sg2/Tests/Generation/GenerationTests.cs:46]: Some engineers have found it useful to know how to get the information provided, but that could be done in a separate helper function somewhere (I have no strong opinions)

**Comment by esmelusina** [comment]
This introduces a regression with SubGraphs, which cause some of the user added referables to disappear; however, I'm greening this as an acceptable regression for now because the current context nodes that SubGraphs are using are getting deprecated and this issue will probably be addressed when I PR context node initialization changes in assets.

**Review by esmelusina — APPROVED** [review]

**Pull request merged**